pytensor/link/numba/dispatch/tensor_basic.py

```diff
@@ -150,14 +139,54 @@ def split(tensor, axis, indices):
 
 
 @numba_funcify.register(ExtractDiag)
-def numba_funcify_ExtractDiag(op, **kwargs):
+def numba_funcify_ExtractDiag(op, node, **kwargs):
     offset = op.offset
-    # axis1 = op.axis1
-    # axis2 = op.axis2
+    axis1, axis2 = normalize_axis_tuple((op.axis1, op.axis2), node.inputs[0].type.ndim)
 
-    @numba_basic.numba_njit(inline="always")
-    def extract_diag(x):
-        return np.diag(x, k=offset)
+    if axis1 > axis2:
+        axis1, axis2 = axis2, axis1
+        offset = -offset
+
+    view = op.view
+
+    if axis1 == 0 and axis2 == 1 and node.inputs[0].type.ndim == 2:
+
+        @numba_basic.numba_njit(inline="always")
+        def extract_diag(x):
+            out = np.diag(x, k=offset)
+
+            if not view:
+                out = out.copy()
+
+            return out
+
+    else:
+        axis1p1 = axis1 + 1
+        axis2p1 = axis2 + 1
+        leading_dims = (slice(None),) * axis1
+        middle_dims = (slice(None),) * (axis2 - axis1 - 1)
+
+        @numba_basic.numba_njit(inline="always")
+        def extract_diag(x):
+            if offset >= 0:
+                diag_len = min(x.shape[axis1], max(0, x.shape[axis2] - offset))
+            else:
+                diag_len = min(x.shape[axis2], max(0, x.shape[axis1] + offset))
```

Comment:
Yeah that looks more clean

Reply:
Wait how would that work? I need `i` in the slices?

Reply:
Yeah it's in the list comp that builds the slices. I tested that it worked, did I miss something? Technically it adds another loop so it might not be worth.

Reply:
I didn't see the list comprehension. I think the double loop is less readable then and might increase the chance that numba doesn't fuse the loop